product/scanning.py

```diff
@@ -22,16 +22,50 @@
     AnalysisResult,
     TriageScope,
 )
+from concourse.model.base import (
+    AttribSpecMixin,
+    AttributeSpec,
+)
 from util import not_none, warning, check_type, info, urljoin
 from container.registry import retrieve_container_image, publish_container_image
 from .model import ContainerImage, Component, UploadResult, UploadStatus
 
 
-class ProcessingMode(Enum):
+
+class ProcessingMode(AttribSpecMixin, Enum):
     UPLOAD_IF_CHANGED = 'upload_if_changed'
     RESCAN = 'rescan'
     FORCE_UPLOAD = 'force_upload'
 
+    @classmethod
+    def _attribute_specs(cls):
+        return (
+            AttributeSpec.optional(
+                name=cls.UPLOAD_IF_CHANGED.value,
+                default=True,
+                doc='''
+                    upload the container images. This will :strong:`not` upload the images if they
+                    are already present. Scans will be triggered only for images that are uploaded.
+                ''',
+                type=str,
+            ),
+            AttributeSpec.optional(
+                name=cls.RESCAN.value,
+                default=False,
+                doc='''
+                    trigger a (re)scan of container images. Images will be uploaded if they are not
```

Comment:
nit: unless they are..

Reply:
also: what precisely was the difference between UPLOAD_IF_CHANGED and RESCAN again? I don't remember. and I do not understand the difference from the (new) documentation tbh

Reply:
UPLOAD_IF_CHANGED uploads and, thus, rescans only those images whose image descriptor changed. RESCAN causes all given container images to be rescanned, uploading those that are not available anymore.

I've tried to clarify it a bit.